File: old/sorting/buy_choco.js
var buyChoco = function(prices, money) {
  prices.sort((a,b) => a-b);
  if(money - prices[0] - prices[1] < 0) {
      return money;
  } else {
      return money - prices[0] - prices[1];
  }
};

console.log(buyChoco([1,2,2], 3));

File: 2025/greedy/buy_choco.py
from typing import List


class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        prices.sort()
        start = money
        cnt = 0
        for price in prices:
            if start-price >= 0:
                start -= price
                cnt += 1
            if cnt == 2:
                break
        return start if cnt == 2 else money
    
print(Solution().buyChoco([1,2,2], 3))